app/uk/gov/hmrc/bindingtariffclassification/model/Search.scala

```diff
@@ -124,6 +131,7 @@ object Filter {
       Some(
         Right(
           Filter(
+            applicationType = param(applicationTypeKey).map(bindApplicationType).filter(_.isDefined).map(_.get),
```

Comment:
Looks isDefined is what a map does implicitly when you try to map it or flatten it. Could this next line works instead?  **param(applicationTypeKey).map(bindApplicationType).flatten**

Reply:
Think a flatMap is what it needed